### DISCUSSION TITLE
Get image width/height

### DISCUSSION THREAD

**Discussion by probablykasper** [supabase/supabase#11822] [Feature Requests]
I'd like to get the image width/height of files in Supabase Storage.

In my case, I'm storing image URLs in my database, and would like to do a migration to also store the image dimensions. It would be great to fill in them server-side, but I believe that would require downloading all the images and using some npm package to get the image dimensions. It would be great to have a more simpler way.

**Comment by GaryAustin1** [comment]
There is a metadata column in the storage.objects table that contains the size metadata of the file.

**Reply by probablykasper** [reply]
Sorry should've clarified I mean image width/height, not file size

**Reply by GaryAustin1** [reply]
Yeah, that is different.  Wonder if the new image transformations could have an option to return the size.
https://github.com/supabase/supabase/discussions/10918

**Reply by probablykasper** [reply]
Looked into that, and I don't think that would work because the transforms are part of `getPublicUrl`, which is a synchronous function that just returns a url string.

**Reply by GaryAustin1** [reply]
Ah, yeah.  Did not think that thru at all.  Transforms are just parameters in the URL.  Sorry for the dead end.

**Reply by probablykasper** [reply]
All good, I didn't think about that at first either
